### ISSUE TITLE
FatalExecutionEngineError when accessing setter of a struct

### ISSUE THREAD

**Issue by dorianru** [ArxOne/MrAdvice#108]
Upon accessing the setter of a struct, a FatalExecutionEngineError (error code 0xc0000005) occurs.
I am using MrAdvice 2.5.1 with .NET Framework 4.

MWE:
```csharp
using System;
using ArxOne.MrAdvice.Advice;
using MrAdviceTest;

[assembly: TestAdvice]

namespace MrAdviceTest
{
	public class TestAdvice : Attribute, IMethodAdvice
	{
		public void Advise(MethodAdviceContext context)
		{
			context.Proceed();
		}
	}

	public struct TestStruct
	{
		public int N { get; set; }
	}

	public class Program
	{
		static void Main(string[] args)
		{
			var t = new TestStruct();
			t.N = 12;
		}
	}
}
```

**Comment by picrap** [comment]
Sorry to read this, I'll take a look at it quickly.

**Comment by picrap** [comment]
Fixed in 2.5.2
